### ISSUE TITLE
[Snowflake] LATERAL FLATTEN gives unexpected parsing errors

### ISSUE THREAD

**Issue by manish-ramrakhiani** [sqlfluff/sqlfluff#2909]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I am using dbt template with dialect Snowflake.
I have used a valid LATERAL FLATTEN in one of the dbt models that compiles successfully and is also executed successfully in dbt. However, sqlfluff throws errors which seems that it is not able to parse and interpret the LATERAL FLATTEN usage correctly.

Here is the piece of code
```
 select
        u.dwhid,
        ura.value::string as user_role
    from {{ ref('stg_users') }} as u,
    LATERAL FLATTEN (INPUT => u.ROLES) ura
```

and here are the sqlfluff errors I am getting
<img width="613" alt="Screenshot 2022-03-24 at 09 17 15" src="https://user-images.githubusercontent.com/93719149/159872232-5cfab40c-8c8c-4f66-ad1f-3826001dc163.png">

I have tried running the plain sql without any jinja templates as well and I still get the same issues.

### Expected Behaviour

Sqlfluff lint should parse the LATERAL FLATTEN correctly and not throw any errors related to the correct usage of this

### Observed Behaviour

Because the actual query runs just fine. Sqlfluff is incorrectly interpreting the usage of LATERAL FLATTEN

### How to reproduce

Just create a table with one of the columns ad Variant and try to use the LATERAL FLATTEN to unnest the Variant 

### Dialect

Snowflake

### Version

0.11.0

### Configuration

.sqlfluff
```
[sqlfluff]
templater = dbt
exclude_rules = L031, L009

[sqlfluff:templater:dbt]
project_dir = ..
profile = default
target = dev

[sqlfluff:rules]
max_line_length = 100

[sqlfluff:rules:L016]
# Line length
ignore_comment_lines = true
ignore_comment_clauses = true

[sqlfluff:rules:L059]
# Policy on quoted and unquoted identifiers
prefer_quoted_identifiers = False
ignore_words = From,To
```

.sqlfluffignore
```
target/
# dbt <1.0.0
dbt_modules/
# dbt >=1.0.0
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
You need to set the dialect to `snowflake` in your `.sqlfluff` and then it parses fine:

```
[sqlfluff]
dialect = snowflake
```

With it using the default `ansi` I get same errors as you above.
